drivers/serial/uart_native_posix.c

```diff
@@ -261,9 +266,20 @@ static void np_uart_poll_out(const struct device *dev,
 				      unsigned char out_char)
 {
 	int ret;
-	struct native_uart_status *d;
+	struct native_uart_status *d = (struct native_uart_status *)dev->data;
+	struct pollfd pfd = { .fd = d->out_fd, .events = POLLHUP };
+
+	if (wait_pts) {
```

Comment:
Yes it is to catch disonnects, not just first opening.

Reply:
What is the use case? It is quite a bit of overhead polling for each char..

Reply:
It actually poll without timeout (the third argument is 0) if slave is connected, so basically it returns immediately if POLLHUP is set. Also, we should consider this kind of an 'overhead' on native posix platform accetable. 
General use case is to be able to receive UART output (which may be used be any Zephyr subsystem or application part) when pts is not yet attached, because without waiting, it is then lost completely (as it is not buffered).